### DISCUSSION TITLE
Some Mining-Related Feedback/Suggestions

### DISCUSSION THREAD

**Discussion by Hecter94** [endless-sky/endless-sky#9888] [Balance, Mechanics, and Interactions]
I recently finished a mining-focused run, stretching from the beginning of the game to the end of FW. I have minimal experience with mining gameplay even before the recent rebalance, so consider this the perspective of a player new to the mechanic. 
My ship progression was as follows:
Sparrow -> Fury -> Sunder -> Argosy -> Bastion -> Behemoth -> Bactrian. 

I had a hard time getting started with mining. Initially, I couldn't afford any mining equipment and didn't know where to find valuable ores, so I mainly mined low-value ores like silicon or iron. I had to collect the flotsam manually and then sell only 15t at a time, barely covering my mortgage expenses. This early stage of the game was quite a slog and not very enjoyable.

However, after paying off my loan and acquiring some mining-related equipment, such as an asteroid scanner, a cargo expansion, and a few mining lasers, I found mining extremely profitable. With my Fury, I was making around $300,000 per trip, and with my Behemoth, I was making around $3,000,000. Mining proved far more profitable than straight trading or running jobs and missions. 

Strangely, I never needed more than a single tractor beam, from the tiny Sunder to the massive Bactrian. Even with a full complement of nine mining lasers, one tractor beam was sufficient to cover all of them. 

Mining seemed like a very safe method of generating income since I could bribe pirates and stay safely in dangerous systems for as long as necessary to mine a full cargo hold. 

However, it was also very static since I was always mining in Orbana and selling in Tania Australis. I never found a reason to mine or sell anywhere else. This was unlike doing jobs or running trade routes which encourage you to move around the galaxy, finding the best trade routes, or going wherever the job board takes you.

Some things that I think would improve it:

- I think the mining lasers could use some more range. They are not very effective as weapons, so extending their range would not make them overpowered. However, their current range is so short that you can't even get all of your mining lasers to focus on a single asteroid on certain ships with widely spaced hardpoints. Even with smaller,  agile ships, getting into the right range to use the mining lasers effectively can be challenging.

- I also found it odd that the mining laser turret was only a single laser. It would be better as a dual laser turret. That does take away some of its ability to function on very small ships but allows it to scale better onto large ships and helps ships with a lot of weapon space use more of it on mining lasers. 

- There should be some method of pushing the player to change where they are mining/selling. Could minable asteroids take random amounts of time to respawn? Or could repeatedly selling ore at the same location decrease its price, similar to commodities? 

- In a similar vein, mining-related jobs would be neat, such as "source x units of ore and deliver them to y" or "Protect NPC ships while they mine in y," The tutorial did tutorialize a "mine x ore and deliver to y." mission, so it was odd that it never came up again. 

- Currently, less valuable minables tend to break apart into more pieces of flotsam than more valuable minables. Could we reverse this relationship? Having cheap minables break apart into one or two pieces of flotsam would make them much easier to collect manually, reducing some of the early game slog that you need to go through in a mining-focused playthrough while having more valuable minables break apart into large groups of flotsam would incentivize adding more tractor beams as you start upgrading your ships and start venturing out further into unsafe territory while hunting for valuable minables. For example, silicon might break apart into 1-2 chunks of 5-10 silicon, whereas Platinum might break apart into five chunks of 1 platinum.

**Comment by MasterOfGrey** [comment]
> Could we reverse this relationship? Having cheap minables break apart into one or two pieces of flotsam would make them much easier to collect manually, reducing some of the early game slog that you need to go through in a mining-focused playthrough while having more valuable minables break apart into large groups of flotsam would incentivize adding more tractor beams as you start upgrading your ships and start venturing out further into unsafe territory while hunting for valuable minables. For example, silicon might break apart into 1-2 chunks of 5-10 silicon, whereas Platinum might break apart into five chunks of 1 platinum.

This feels like the most valuable/actionable insight here I'd say. Sounds like it'd be a good gameplay change for an extremely minimal effort.

**Comment by VaelophisNyx** [comment]
I do wonder if it'd be possible to offer a starting-cash-capable mining vessel on new boston, for those who want to mine but not slog through to buying a mining laser

**Reply by TheGiraffe3** [reply]
There is a plugin for that.

**Reply by VaelophisNyx** [reply]
well yeah but like, in the context of vanilla

**Comment by yowshi** [comment]
personally i think we need less competition in mining. i love mining but the biggest hassle i am having is within a minute of getting to a system rich in minable asteroids they are all gone, because some big fleet rolls in and mines them all before i can get more then a dozen asteroids mined.

then i gotta land again and launch so the system refills with mineable rocks.

**Comment by samrocketman** [comment]
There's a couple of ways I addressed this in my fork and wanted to address in vanilla.

* Enable your fleet to self-mine without direction when they have asteroid scanners.  This is not much different than your fleet in combat.  I never implemented this idea because it was generally always shot down hard by those that would be involved with merging.
* Make mining drones cheaply available (50-60k credits).  Mining drones take a pretty big load of effort off the player while mining.  A destroyed drone is no big deal because of the cost and it encourages more carrier gameplay.
* Fighter and drone bays as outfits.  This enables non-carriers to be capable of carrying mining drones.  Considering the cheap price of mining drones this enables earlier game gameplay with drones.  Without these kinds of outfits the barrier to entry to mining with a fighter in vanilla is around 7 million credits.  No drones available for player purchase.
* Ore-oriented missions (and perhaps a miners guild).  Different regions might request ore from different reaches of the galaxy where the ore type is locally sparse.  Lucrative missions which require the player to travel would be good.
* Mining survey missions.  These should require the player to travel around surveying systems for minables.  A reward for handing over survey data.
* Outfits which manufacture from minables.  This encourages the player to stock some ore as they travel to create consumables from what is mined.  This idea could branch into the concept of recipes.

**Reply by yowshi** [reply]
yeah but drones dont have cargo capacity. or most dont so they cant retrieve what is mined. which means what is mined is going to be gobbled up by npc fleets.

as for your own fleet. this issue hampers your ability to get one. or it hampers mine anyway.

**Reply by samrocketman** [reply]
A mining drone should have 5 cargo; that's the point of a mining drone.  As for human fighters one way you can try them out in vanilla is use a cargo expansion.

**Reply by asl97** [reply]
@samrocketman 
> * Enable your fleet to self-mine without direction when they have asteroid scanners.

Is that suppose to be phase 2 that you mentioned in https://github.com/endless-sky/endless-sky/pull/6669#issuecomment-1086772214 ?
I guess not, sound slightly different
> * Phase 2: add a fleet command to mine/gather all resources in the current system the player is located.

____
> I never implemented this idea

So neither version is in your fork?

How hard would it be to implement some kind of command to get a sizable fleet to mine stuff?

I brought over 130 fighter for my fleet to only realize for mining I must micro managing them by manually selecting each ship and asteroid, which is just not practical on the mobile port.

I guess something like this pseudo code:
```c++
asteroid = asteroids.begin()
if (asteroid == asteroids.end()){ return } // no astroid in system
for (ship = player_fleet.begin(), ship != player_fleet.end(); ++ship){
  if (!*ship.canTargetAsteroid()) {continue}
  *ship.target(*asteroid)
  ++asteroid
  if (asteroid == asteroids.end()){ asteroid = asteroids.begin() } // more ship than asteroid in system, assign multiple ship to each asteroid
}
```

**Reply by samrocketman** [reply]
Auto-mining was proposed (without code) for phase two but rejected by maintainers.  Bear in mind this was at a time where having your fleet assist you at all with mining was rejected and I had to argue in favor of the feature for a long period of time before it became an acceptable idea (and so the feature would be acceptable to merge). Auto-mining is the same feature mentioned in phase 2 requiring asteroid scanners that I mentioned.  Purchasing asteroid scanners is a makes sense mechanic for that kind of feature I think.  I plan to implement it in [my fork] when I start messing around with ES again because I think you should be able to do that; but it's not something you will likely see in vanilla.

Another valuable/fun addon for drones and fighters in [my fork] is bays as installable outfits which replace guns and turrets.  This idea was actually proposed as code and rejected.  It continues to make for good gameplay which I enjoy.

[my fork]: https://github.com/samrocketman/endless-sky/releases/tag/continuous-with-plugins

**Comment by Quantumshark** [comment]
I'm going to come out and admit that I made the wrong call with balancing prospecting.
Currently, the mining laser has the following effects on mineable drop rates:
<img width="307" height="295" alt="image" src="https://github.com/user-attachments/assets/b72422b5-32c5-4a31-bdcb-3b6810399917" />

I realise now that I've got this backwards. The precision mining from specialised mining weapons should be more useful on higher-value mineables, not the opposite - that way, higher-end mining benefits more from specialising, whereas now it benefits less.

I think what I ought to do is flip the direction of that progression (keeping aluminium the same, since a) that's the baseline for calibrating different prospecting weapons against each other, and b) 20%+ is a good effect size, whereas the <10% increase on good mineables really isn't big enough to be noticeable in normal gameplay), so that silicon gains the least benefit from prospecting (but perhaps has a higher base yield), while the higher value types gain the most.

**Reply by Quantumshark** [reply]
As a proposal for new scaling, how about something like this:
<img width="304" height="317" alt="image" src="https://github.com/user-attachments/assets/c60685e8-45aa-4d10-a4ca-990e3bc53f85" />
